src/OSPSuite.SimModelNative/src/Quantity.cpp

```diff
@@ -47,6 +47,7 @@ void Quantity::DeleteFormula()
 	}
 
 	_originalValueFormula=NULL;
+   _originalFormulaID = INVALID_QUANTITY_ID;
```

Comment:
this is the only fix.
The rest is just adding test C++ application, which can be used for stand alone profiling and tests